### DISCUSSION TITLE
Can I set up a custom provider as a proxy?

### DISCUSSION THREAD

**Discussion by ytoaa** [qdm12/gluetun#935] [Q&A] [answered]
Green tunnel does not have user authentication, so I would like to use it by adding it to gluetun.

**Comment by qdm12** [comment]
Can you provide me more details on what green tunnel is? What protocol does it implement?
What does it add compared to the existing HTTP proxy server and/or Shadowsocks proxy server?

**Reply by ytoaa** [reply]
https://github.com/SadeghHayeri/GreenTunnel It has this technology and acts as a proxy server.

**Comment by qdm12** [comment, marked as answer]
You can use green tunnel as a container and make it use the gluetun container as its network. See https://github.com/qdm12/gluetun/wiki/Connect-a-container-to-gluetun and https://github.com/qdm12/gluetun/wiki/Port-mapping
